### ISSUE TITLE
Build problem on Fedora 20

### ISSUE THREAD

**Issue by dcesari** [apptainer/singularity#56]
I have the following header conflict when building [9f27c61](https://github.com/gmkurtzer/singularity/commit/9f27c61b2a5da92dae15fd135d87141d8dc48a6c) on Fedora 20: 

```
gcc -DHAVE_CONFIG_H -I.  -DARCH_x86_64  -Wall -g -O2 -MT ftrace-ftrace.o -MD -MP -MF .deps/ftrace-ftrace.Tpo -c -o ftrace-ftrace.o `test -f 'ftrace.c' || echo './'`ftrace.c
In file included from ftrace.c:28:0:
/usr/include/linux/ptrace.h:58:8: error: redefinition of ‘struct ptrace_peeksiginfo_args’
 struct ptrace_peeksiginfo_args {
        ^
In file included from ftrace.c:27:0:
/usr/include/sys/ptrace.h:191:8: note: originally defined here
 struct ptrace_peeksiginfo_args

```

solved commenting the line

```
#include <linux/ptrace.h>
```

in `ftrace.c`. Is the fix correct?

**Comment by truatpasteurdotfr** [comment]
gcc -DHAVE_CONFIG_H -I.  -DARCH_x86_64  -Wall -g -O2 -MT ftrace-ftrace.o -MD -MP -MF .deps/ftrace-ftrace.Tpo -c -o ftrace-ftrace.o `test -f 'ftrace.c' || echo './'`ftrace.c
In file included from ftrace.c:28:0:
/usr/include/linux/ptrace.h:58:8: error: redefinition of 'struct ptrace_peeksiginfo_args'
 struct ptrace_peeksiginfo_args {
        ^
In file included from ftrace.c:27:0:
/usr/include/sys/ptrace.h:89:8: note: originally defined here
 struct ptrace_peeksiginfo_args {
        ^

Same here on alpine linux kernel-4.4.11-0-grsec, commenting line 28 fixes the build and allow singularity to be built/run :).

**Comment by loveshack** [comment]
I think ftrace is obsolete, and you could just comment it out of the build (#55).

**Comment by gmkurtzer** [comment]
@loveshack is correct that ftrace is not currently being used in the codebase, but it will be shortly... I have plans for it. lol

@dcesari Yes, I believe you are correct as sys/ptrace.h seems to be the correct header. Fixed in [master c4d4f5d].

Thanks!
